### PR TITLE
feat: Add version formating for modules

### LINKED ISSUE #2576: Use the version format string for all versions
#### Is your feature request related to a problem? Please describe.

In #2499 and #2566 we added the ability to customise how we format the versions this has been added to a few modules (`python`, `node`, `java`, `rust` and `ruby`) but it would be good to add it to all the other modules that display a version.

#### Describe the solution you'd like

All modules that display some kind of version can use the format version.

### PR DESCRIPTION
**Pull request by cymruu**
#### Description
Add support for version formatting for the rest of modules. 
 
#### Motivation and Context
Closes #2576

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/formatter/version.rs
Comment (vladimyr) [inline src/formatter/version.rs:66]: Well, you only need the module name, why not pass that instead?

**Comment by andytom** [comment]
Hi @cymruu, can you fix the merge conflicts and I think we should be about ready to merge this.

**Comment by offbyone** [comment]
Oh, _very_ nice! Thanks for coming in and doing all the work I was too lazy to finish! :D

**Review by andytom — APPROVED** [review]
LGTM, but would like a second opinion since this touches a lot of modules and I might have missed something.

**Comment by offbyone** [comment]
It all looks good to me, FWIW. I think we should let the workflow run to make sure it builds cleanly.

**Comment by andytom** [comment]
> It all looks good to me, FWIW. I think we should let the workflow run to make sure it builds cleanly.

Huh, I'd not seen that before must have been a new change for Github.

**Comment by offbyone** [comment]
It is: https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/

Mainly working to prevent abuse by cryptocurrency miners.

**Comment by offbyone** [comment]
(I'd have hit "go" on that myself, but ... 
image

:D

**Comment by andytom** [comment]
Thank you for your contribution @cymruu and thanks for reviewing @offbyone.

**Comment by andytom** [comment]
> It is: https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/
> 
> Mainly working to prevent abuse by cryptocurrency miners.

Ah ok that explains it. Annoyingly it doesn't show anything in the app.

**Comment by vladimyr** [comment]
This was truly a Herculean effort! :heart: :100: 

I have one small suggestion though. Let's agree on naming. Either we decide to go with `get_something_version` or with `parse_something_version` (personally, I believe the latter is a better name) and then we should ensure that's consistent across all modules.

**Pull request merged**
